source/common/network/filter_manager_impl.cc

```diff
@@ -87,6 +88,7 @@ FilterStatus FilterManagerImpl::onWrite(ActiveWriteFilter* filter,
 
   std::list<ActiveWriteFilterPtr>::iterator entry;
   if (!filter) {
+    connection_.streamInfo().addBytesSent(buffer_source.getWriteBuffer().buffer.length());
```

Comment:
I think you probably actually want this to be data that is written to the connection, so should this happen after final filter iteration below?

Reply:
Excellent point, thanks. I think we want "black box" telemetry: bytes read before transformation and bytes sent after transformation.